### ISSUE TITLE
[mu bug] can't extract message parts of type "message/rfc822"

### ISSUE THREAD

**Issue by p4tpr0** [djcb/mu#2467]
**Describe the bug**
Running from the command line, I can't extract message parts of type "message/rfc822". The error reads "error: not a part"

**To Reproduce**
```
$ mu extract -d /tmp/mymessage.eml
MIME-parts in this message:
  1 <none> text/plain [inline] (6585 bytes)
  2 <none> message/rfc822 [inline]

$ mu extract -d --parts=2 /tmp/mymessage.eml
error: not a part
usage: ../..
```

**Environment**
I'm running mu 1.8.13 on FreeBSD 13.1.
Previously (yesterday), the same command on mu 1.2 running on FreeBSD 11.4 was working without error.

**Checklist**
- [x] you are running either the latest 1.8.x/1.10.x release or `master` (otherwise, please upgrade).
I'm running the latest FreeBSD package available and I've compiled mu 1.8.14: same result.
[mymessage.eml.zip](https://github.com/djcb/mu/files/11150299/mymessage.eml.zip)

**Comment by p4tpr0** [comment]
In the mean time, I've extracted the binary from FreeBSD package for version 1.2 -> no error.

**Comment by djcb** [comment]
Fixed in master, and should make it to 10.2 when that is released.
